internal/publish/acker.go

```diff
@@ -70,6 +70,8 @@ func (w *WaitPublishedAcker) Close() {
 	w.decref(1)
 }
 
+func (w *WaitPublishedAcker) ClientClosed() {}
```

Comment:
Should this be decrementing the reference count? Maybe this is why the benchmark is hanging...?

Reply:
Good catch! That's the issue!